src/runtime/api-channel.ts

```diff
@@ -14,46 +14,39 @@ import {Arc} from './arc.js';
 import {DevtoolsConnection} from './debug/devtools-connection.js';
 import {OuterPortAttachment} from './debug/outer-port-attachment.js';
 import {Handle} from './handle.js';
-import {ParticleSpec, SerializedParticleSpec} from './particle-spec.js';
+import {ParticleSpec} from './particle-spec.js';
 import {Particle} from './particle.js';
 import * as recipeHandle from './recipe/handle.js';
 import * as recipeParticle from './recipe/particle.js';
 import {StorageProxy} from './storage-proxy.js';
 import {SerializedModelEntry} from './storage/crdt-collection-model.js';
 import {StorageProviderBase} from './storage/storage-provider-base.js';
-import {Type, TypeLiteral} from './type.js';
-import {PropagatedException, SerializedPropagatedException} from './arc-exceptions.js';
+import {Type} from './type.js';
+import {PropagatedException} from './arc-exceptions.js';
+import {Literal} from './hot.js';
 
 enum MappingType {Mapped, LocalMapped, RemoteMapped, Direct, ObjectMap, List, ByLiteral}
 
-interface MappingInfo {
+interface MappingInfo<T> {
   type: MappingType;
   initializer?: boolean;
   redundant?: boolean;
-  value?: MappingInfo;
-  key?: MappingInfo;
-  converter?: Literalizer | LiteralizerParticleSpec | LiteralizerPropagatedException;
+  value?: MappingInfo<T>;
+  key?: MappingInfo<T>;
+  converter?: Literalizable<T, Literal>;
   identifier?: boolean;
   ignore?: boolean;
 }
 
-// TODO(shans): are there better types that I can use for this?
-interface Literalizer {
-  prototype: {toLiteral: () => TypeLiteral};
-  fromLiteral: (typeliteral: TypeLiteral) => Type;
-}
-
-interface LiteralizerParticleSpec {
-  prototype: {toLiteral: () => SerializedParticleSpec};
-  fromLiteral: (spec: SerializedParticleSpec) => ParticleSpec;
-}
-
-interface LiteralizerPropagatedException {
-  prototype: {toLiteral: () => SerializedPropagatedException};
-  fromLiteral: (exception: SerializedPropagatedException) => PropagatedException;
+interface Literalizable<T, Lit extends Literal> {
+  prototype: {
+    toLiteral(): Lit;
+  };
+  fromLiteral(literal: Lit): T;
 }
 
-const targets = new Map<{}, Map<string, MappingInfo[]>>();
+// tslint:disable-next-line:no-any
+const targets = new Map<{}, Map<string, MappingInfo<any>[]>>();
```

Comment:
would the new `never` type work better here? Anything can become `never`, but `never` can't be treated as anything without an explicit cast.

Reply:
Do you mean `unknown`? 

https://www.typescriptlang.org/docs/handbook/release-notes/typescript-3-0.html#new-unknown-top-type